birdhouse/components/cowbird/docker-compose-extra.yml

```diff
@@ -23,7 +23,11 @@ services:
       # root user
       COWBIRD_FILESYSTEM_ADMIN_UID: 0
       COWBIRD_FILESYSTEM_ADMIN_GID: 0
-      WORKSPACE_DIR: /${USER_WORKSPACES}
+      # Note that WPS_OUTPUTS_DIR and WORKSPACE_DIR must both point to paths from the same volume.
+      # This is to allow the creation of hardlinks between the wpsoutputs and the user workspace.
+      WPS_OUTPUTS_DIR: ${WPS_OUTPUTS_DIR}
+      WPS_OUTPUTS_PUBLIC_SUBDIR: ${WPS_OUTPUTS_PUBLIC_SUBDIR}
+      WORKSPACE_DIR: ${DATA_PERSIST_ROOT}/${USER_WORKSPACES}
```

Comment:
Are the variables themselves actually used in Cowbird (through `os.getenv` or similar)? 
It seems they should already be included in the [cowbird/config.yml.template](https://github.com/bird-house/birdhouse-deploy/pull/360/files#diff-0752f5d9727dae924cef084a00c6b65c26945b4a08dd681bb0d6cf702b68d33e), but maybe I'm mistaken? 
Not a big deal though if they are provided again here.

Reply:
The variables are indeed also included in the `config.yml.template` file, but when that template file gets converted to `config.yml` file, the env variables are not all resolved. I notice for example that the `WORKSPACE_DIR` and `WPS_OUTPUTS_DIR` keep their unresolved format `${VAR_NAME}`. By including the environment variables here in the `docker-compose`, the variables can get resolved later in Cowbird if necessary. 

When loading the config in Cowbird, it attempts to resolved remaining unresolved env vars found in the config 
 using `os.path.expandvars()`.